dotnet-gitchanges/Readers/Parsers/GitTagCommitParser.cs

```diff
@@ -0,0 +1,62 @@
+using System;
+using System.Text.RegularExpressions;
+using System.Linq;
+using Gitchanges.Changes;
+using Gitchanges.Configuration;
+using LibGit2Sharp;
+
+namespace Gitchanges.Readers.Parsers
+{
+    public class GitTagCommitParser : ICommitParser<IChange>
+    {
+        private const string Unreleased = "Unreleased";
+        private readonly ParsingPatterns _patterns;
+        private string _lastVersion = Unreleased;
+        private readonly IRepository _repository;
+        
+        public GitTagCommitParser(ParsingPatterns patterns, IRepository repository)
+        {
+            _repository = repository;
+            _patterns = patterns;
+        }
+        
+        public IChange Parse(Commit commit)
+        {
+            var gitTag = _repository.Tags.FirstOrDefault(t => t.Target.Sha == commit.Sha);

```

Comment:
We can avoid iterating through the tags on every commit if we have a map of Sha -> Tag to hit

Reply:
Will do that